staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go

```diff
@@ -84,7 +84,17 @@ func (c *azureTokenCache) setToken(tokenKey string, token *azureToken) {
 	c.cache[tokenKey] = token
 }
 
+var warnOnce sync.Once
+
 func newAzureAuthProvider(_ string, cfg map[string]string, persister restclient.AuthProviderConfigPersister) (restclient.AuthProvider, error) {
+	// deprecated in v1.22, remove in v1.25
+	warnOnce.Do(func() {
+		klog.Warningf(`WARNING: the azure auth plugin is deprecated in v1.22+, unavailable in v1.25+; use https://github.com/Azure/kubelogin client-go credential plugin instead.
```

Comment:
should this get the same V(1) treatment and move to a Warningf in 1.24?

Reply:
Pinged @ritazh @aramase @weinong @feiskyer @karataliu @andyzhangx

Reply:
+1 on what @liggitt pointed out.

Reply:
Fixed.  Opened #103525 to track.

Reply:
looks good.